### PR TITLE
Add binary build

### PR DESCRIPTION
**Pull request by mthrok**

### CODE REVIEW

**Comment by mthrok** [comment]
@zhangguanheng66 I ported binary build job from torchaudio. The benefit of this is that after conda package is built, it performs import test. So the concern regarding C++ extension proper installation is automatically addressed.

File: packaging/torchtext/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    - requests
+    - tqdm
+    {{ environ.get('CONDA_PYTORCH_CONSTRAINT') }}
+
+build:
+  string: py{{py}}
+
+test:
+  imports:
+    - torchtext
```
Comment (mthrok) [inline packaging/torchtext/meta.yaml:31]: @zhangguanheng66 This is where the import test is defined.

Comment (zhangguanheng66) [reply packaging/torchtext/meta.yaml:31]: This is the old conda build script with some import tests [link](https://github.com/pytorch/text/blob/32bc374d7d8b15ed431dfaa64304a974a959ea5e/build_tools/conda/torchtext/meta.yaml#L25)

Comment (mthrok) [reply packaging/torchtext/meta.yaml:31]: Oh interesting. why was it removed?

File: .circleci/config.yml.in [hunk 2]
```diff
@@ -287,13 +290,13 @@ jobs:
...
-  #     - circleci_consistency
-  #     {{ build_workflows() }}
-  #     - binary_win_conda:
-  #         name: torchtext_win_py3.6
-  #         python_version: "3.6"
+  build:
+    jobs:
+      - circleci_consistency
+      {{ build_workflows() }}
+      #  - binary_win_conda:
```
Comment (cpuhrsch) [inline .circleci/config.yml.in:297]: Is this to be fixed in a follow-up PR?

Comment (mthrok) [reply .circleci/config.yml.in:297]: Not my intent.
I am just keeping what was there and making `build_workflows` work.

Comment (zhangguanheng66) [reply .circleci/config.yml.in:297]: Should we test the torchtext package with cpp extension on a Windows machine? Although we don't explicitly support Windows, the current package does work on Windows.

Comment (mthrok) [reply .circleci/config.yml.in:297]: Ideally, we should. `torchaudio` does not have this but needs this soon.
We can look into how `torchvision` does it or ask help from someone with Windows experience.

Comment (zhangguanheng66) [reply .circleci/config.yml.in:297]: @peterjc123

Comment (peterjc123) [reply .circleci/config.yml.in:297]: I'll add unit tests first. Binary builds on Windows will come later.

**Pull request merged**